server/controllers/userController.js

```diff
@@ -1,63 +1,40 @@
+/* eslint-disable no-console */
 /* eslint-disable max-len */
 import hash from 'bcrypt-nodejs';
 import dotenv from 'dotenv';
 import uuid from 'uuid';
-import User from '../models/users';
-import userSchema from '../validations/userValidation';
-import helperFunction from '../helpers/helperFunction';
-import successResponse from '../helpers/successResponse';
+import db from '../models/index';
 import failureResponse from '../helpers/failureResponse';
+import successResponse from '../helpers/successResponse';
 import tokenGenerator from '../helpers/userTokenGenerator';
-import db from '../models/index';
 
 dotenv.config();
 
 class userController {
-  /**
-   *@description Displays a welcome message
-   * @param {object} req
-   * @param {object} res
-   * @returns {object} response
-   */
   static entryMessage(req, res) {
     return res.status(200).json({
       status: 200,
       message: 'welcome to MyDiary application',
     });
   }
 
-  /**
-   *@description Creates a user account
-   * @param {object} req
-   * @param {object} res
-   * @returns {object} response
-   */
-
   static async createAccount(req, res) {
     const user = req.body;
-    user.password = hash.hashSync(user.password);
     const {
       firstname, lastname, email, password,
     } = user;
-    const validateUser = userSchema.validate({
-      firstname: user.firstname, lastname: user.lastname, email: user.email, password: req.body.password,
-    });
-    if (validateUser.error) {
-      return failureResponse(res, 400, validateUser.error.details[0].message);
-    }
+    user.password = hash.hashSync(user.password);

```

Comment:
I would like to understand why you are hashing the password here and also inline 32

Reply:
I think that is how bcrypt-nodejs works. because others here are using bcrypt or bcryptjs which I couldn't get to work. That is why I use bcrypt-nodejs and it seems like for it to work on login I need to hash twice.

Reply:
Alright no problem